From 6972aec64a9104652fe372720210308ae64c8c2b Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Sat, 04 Jan 2025 10:49:57 +0200
Subject: [PATCH] [FSW-11035] fs1: jtag disable control reg update

With specs_h15 1p33, jtag disable register name
updated. Appropriate changes are taken care
as part of this patch.

Change-Id: Icbff48797dbbc3f182bda9b14ed9765f089b324d
---

diff --git a/arch/fs1/arc/dev_security.c b/arch/fs1/arc/dev_security.c
index e5ebf46..e36ef5d 100644
--- a/arch/fs1/arc/dev_security.c
+++ b/arch/fs1/arc/dev_security.c
@@ -475,11 +475,11 @@
 
 void security_handle_jtag_if(void)
 {
-	uint32_t val = regmap_readl(IO0_H0_PSOC_BOOT(mmPSOC_BOOT_CONF_JTAG_DISABLE));
+	uint32_t val = regmap_readl(IO0_H0_PSOC_BOOT(mmPSOC_BOOT_CONF_JTAG_DISABLE_CTRL));
 
 	if ((boot_rom_global.config_fuse & EFUSE_SYSCFG_DIS_JTAG) == 0) {
 		/* Enable JTAG */
-		val &= ~(PSOC_BOOT_CONF_JTAG_DISABLE_VAL_M);
+		val &= ~(PSOC_BOOT_CONF_JTAG_DISABLE_CTRL_DISABLE_M);
 
 		security_rr_jtag_config(true);
 
@@ -488,7 +488,7 @@
 		 * Already all PBs are opened for RW in all pages for all SAI levels.
 		 */
 	} else {
-		val |= PSOC_BOOT_CONF_JTAG_DISABLE_VAL_M;
+		val |= (PSOC_BOOT_CONF_JTAG_DISABLE_CTRL_DISABLE_M);
 
 		security_rr_jtag_config(false);
 
@@ -499,9 +499,9 @@
 	}
 
 	/* Clear FSM timer */
-	val &= ~(PSOC_BOOT_CONF_JTAG_DISABLE_COUNTER_EN_M);
+	val &= ~(PSOC_BOOT_CONF_JTAG_DISABLE_CTRL_COUNTER_EN_M);
 
-	regmap_writel(val, IO0_H0_PSOC_BOOT(mmPSOC_BOOT_CONF_JTAG_DISABLE));
+	regmap_writel(val, IO0_H0_PSOC_BOOT(mmPSOC_BOOT_CONF_JTAG_DISABLE_CTRL));
 }
 
 static uint32_t prepare_mask(uint8_t min, uint8_t max)
